feature/notification/src/main/java/io/github/droidkaigi/confsched2019/notification/service/FCMService.kt

```diff
@@ -0,0 +1,93 @@
+package io.github.droidkaigi.confsched2019.notification.service
+
+import android.app.PendingIntent
+import android.content.Intent
+import android.os.Build
+import androidx.core.os.bundleOf
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import io.github.droidkaigi.confsched2019.ext.android.queryIntentAllActivities
+import io.github.droidkaigi.confsched2019.notification.NotificationChannelInfo
+import io.github.droidkaigi.confsched2019.notification.NotificationUtil
+import io.github.droidkaigi.confsched2019.notification.createDefaultNotificationChannel
+import io.github.droidkaigi.confsched2019.timber.error
+import timber.log.Timber
+
+class FCMService : FirebaseMessagingService() {
+    override fun onNewToken(token: String) {
+        // Need to have channels before receiving channels
+        // So here is a good timing to create them
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            enumValues<NotificationChannelInfo>().forEach {
```

Comment:
I don't know enumValues. I'm just curious. Why you use enumValues instead of enum.values()?

Reply:
That's why I'd like to prefer unifying the declaration style (not only for enum classes, just my preference). We can use `NotificationChannelInfo.values()` as well of course.

As of the raison d'être, please read https://github.com/Kotlin/KEEP/blob/master/proposals/generic-values-and-valueof-for-enums.md